core/src/test/scala/unit/kafka/integration/KafkaServerTestHarness.scala

```diff
@@ -142,6 +127,19 @@ abstract class KafkaServerTestHarness extends QuorumTestHarness {
     super.tearDown()
   }
 
+  def recreateBrokers(reconfigure: Boolean = false, startup: Boolean = false): Unit = {
+    if (reconfigure) {
+      instanceConfigs = null
+    }
+    if (configs.isEmpty)
+      throw new KafkaException("Must supply at least one server config.")
+
+    TestUtils.shutdownServers(_brokers, deleteLogDirs = false)
```

Comment:
Can we set the "alive" array to all false after this line, so that it stays in sync with what is going on?